webapp/app/js/compare.js

```diff
@@ -0,0 +1,494 @@
+                     cmap['deltaPercentage'] = (cmap['delta'] / cmap['originalGeoMean'] * 100).toFixed(2) + ' %';
+
+                    cmap.type = 'data';
+                    //TODO: some tests are reverse, figure out how to account for that
+                    if (cmap.delta < 0) {
+                      cmap.type = 'improvement';
+                    } else if (cmap.delta > 1) {
+                      cmap.type = 'regression';
+                    }
+
+                    //TODO: zoom?  >1 highlighted revision?
+                    var originalSeries = encodeURIComponent(JSON.stringify(
+                                  { project: $scope.originalProject,
+                                    signature: seriesList[idx].signature,
+                                    visible: true}));
+
+                    var newSeries = encodeURIComponent(JSON.stringify(
+                                  { project: $scope.newProject,
+                                    signature: new_seriesList[idx].signature,
+                                    visible: true}));
+
+                    var detailsLink = thServiceDomain + '/perf.html#/graphs?timerange=' +
+                        $scope.timeRange + '&series=' + newSeries;
+
+                    if (seriesList[idx].signature != new_seriesList[idx].signature) {
+                      detailsLink += '&series=' + originalSeries;
+                    }
+
+                    detailsLink += '&highlightedRevision=' + $scope.newRevision;
+
+                    cmap.detailsLink = detailsLink;
+                    cmap.name = seriesList[idx].platform;
+                    cmap.isHeader = false;
+                    compareResultsMap[counter++] = cmap;
+                  }
+                };
+              }
+              $scope.compareResults = Object.keys(compareResultsMap).map(function(k) {
+                return compareResultsMap[k];
+              });
+
+
+              });
+            });
+          });
+        }
+      );
+    }
+
+    function verifyOriginalRevision() {
+      $scope.originalResultSetID = '';
+      var uri = thServiceDomain + '/api/project/' + $scope.originalProject +
+          '/resultset/?format=json&full=false&with_jobs=false&revision=' + 
+          $scope.originalRevision;
+
+      $http.get(uri).then(function(response) {
+        var results = response.data.results;
+        if (results.length > 0) {
+          $scope.originalResultSetID = results[0].id;
+        }
+      });
+    }
+
+    function verifyNewRevision() {
+      $scope.newResultSetID = '';
+      var uri = thServiceDomain + '/api/project/' + $scope.newProject +
+          '/resultset/?format=json&full=false&with_jobs=false&revision=' + 
+          $scope.newRevision;
+
+      $http.get(uri).then(function(response) {
+        var results = response.data.results;
+        if (results.length > 0) {
+          $scope.newResultSetID = results[0].id;
+        }
+      });
+    }
```

Comment:
verifyOriginal and verifyNew are almost the same function. You should refactor them into one and call it twice.